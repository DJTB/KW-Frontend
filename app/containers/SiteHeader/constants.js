import { fromJS } from 'immutable';
/* eslint-disable import/prefer-default-export */
export const navRoutes = fromJS([
  {
    text: 'Reviews',
    to: '/review',
  }, {
    text: 'Vocabulary',
    to: '/vocabulary',
  }, {
    text: 'Settings',
    to: '/settings',
  }, {
    text: 'About',
    to: '/about',
  }, {
    text: 'Contact',
    to: '/contact',
  }, {
    text: 'Logout',
    to: '/logout',
  },
]);
/* eslint-enable */
