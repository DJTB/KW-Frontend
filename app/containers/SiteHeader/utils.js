
/**
 * Add review count to review route in first index
 * Apologies for dodginess
 * @param {Immutable List} list of routes as Maps
 * @param {Number} count
 * @return {Immutable List} new list with count added
 */
export function addReviewCount(list, count) { // eslint-disable-line import/prefer-default-export
  return list.setIn([0, 'count'], count);
}
