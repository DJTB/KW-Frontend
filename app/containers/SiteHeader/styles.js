import styled from 'styled-components';
import { white, whiteDark } from 'shared/styles/colors';
import { adjustColor } from 'shared/styles/utils';
import { bottomLight } from 'shared/styles/shadows';
import { padding } from 'shared/styles/sizing';
import Container from 'components/Container';
import Element from 'components/Element';

export const StyledHeader = styled.header`
  position: relative;
  background-color: ${white};
  box-shadow: ${bottomLight};
  border-bottom: 1px solid ${adjustColor(whiteDark, 'alpha(0.5)')};
`;

export const StyledContainer = styled(Container)`
  /* header already has a lot of space, let's have less padding than most other containers */
  padding-top: ${padding.mobile.inner.y}rem;
  padding-bottom: ${padding.mobile.inner.y}rem;
`;

export const StyledElement = styled(Element)`
  /* logo has enough vertical spacing already */
  padding-top: 0;
  padding-bottom: 0;
`;
