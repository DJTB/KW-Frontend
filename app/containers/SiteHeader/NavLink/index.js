import React from 'react';
import PropTypes from 'prop-types';

import { Li, Link, LinkButton, Text, Count } from './styles';

NavLink.propTypes = {
  text: PropTypes.string.isRequired,
  route: PropTypes.string.isRequired,
  isOffCanvas: PropTypes.bool.isRequired,
  handleLogout: PropTypes.func.isRequired,
  count: PropTypes.number,
};

NavLink.defaultProps = {
  count: 0,
};

function NavLink({
  text,
  route,
  count,
  isOffCanvas,
  handleLogout,
}) {
  return (
    <Li isOffCanvas={isOffCanvas}>
      {text === 'logout' ? (
        <LinkButton onClick={handleLogout}>
          <Text>
            {text}
          </Text>
        </LinkButton>
      ) : (
        <Link
          plainLink
          to={route}
        >
          <Text>
            {text}
            {count > 0 && <Count disabled={count < 1}>{count}</Count>}
          </Text>
        </Link>
      )}
    </Li>
  );
}

export default NavLink;
