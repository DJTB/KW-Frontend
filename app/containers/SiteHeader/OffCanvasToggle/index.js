import React from 'react';
import PropTypes from 'prop-types';
import { branch, renderNothing } from 'recompose';

import { Toggle, ToggleDot } from './styles';

OffCanvasToggle.propTypes = {
  isActive: PropTypes.bool,
  ariaControls: PropTypes.string.isRequired,
  handleToggle: PropTypes.func.isRequired,
};

OffCanvasToggle.defaultProps = {
  isActive: false,
};

function OffCanvasToggle({ isActive, handleToggle, ariaControls }) {
  return (
    <Toggle
      type="button"
      aria-label="navigation menu"
      aria-controls={ariaControls}
      aria-expanded={isActive}
      onClick={handleToggle}
    >
      <ToggleDot />
      <ToggleDot />
      <ToggleDot />
    </Toggle>
  );
}

const isHidden = ({ isVisible }) => !isVisible;
const enhance = branch(
  isHidden,
  renderNothing,
);

export default enhance(OffCanvasToggle);
