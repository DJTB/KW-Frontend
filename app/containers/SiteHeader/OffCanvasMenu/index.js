import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import NavLink from '../NavLink';

import { Wrapper, Ul, CloseButton } from './styles';

OffCanvasMenu.propTypes = {
  links: PropTypes.array.isRequired,
  isVisible: PropTypes.bool.isRequired,
  handleLogout: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
};

function OffCanvasMenu({ links, isVisible, handleLogout, handleClose }) {
  return (
    <Wrapper isVisible={isVisible}>
      <CloseButton onClick={handleClose} name="CLOSE" title="Close menu" size="3rem" />
      <Ul>
        {links.map((link) => (
          <NavLink
            key={cuid()}
            handleLogout={handleLogout}
            isOffCanvas
            {...link}
          />
        ))}
      </Ul>
    </Wrapper>
  );
}

export default OffCanvasMenu;
