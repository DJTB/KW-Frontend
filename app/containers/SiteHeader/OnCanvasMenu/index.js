import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import NavLink from '../NavLink';
import { Ul } from './styles';

OnCanvasMenu.propTypes = {
  links: PropTypes.array.isRequired,
  handleLogout: PropTypes.func.isRequired,
};

function OnCanvasMenu({ links, handleLogout }) {
  return (
    <Ul>
      {links.map((link) => (
        <NavLink
          key={cuid()}
          handleLogout={handleLogout}
          isOffCanvas={false}
          {...link}
        />
      ))}
    </Ul>
  );
}

export default OnCanvasMenu;
