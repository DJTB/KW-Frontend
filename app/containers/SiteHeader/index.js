import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import debounce from 'lodash/debounce';
import { clearToken } from 'utils/auth';
import { breakpoints } from 'shared/styles/media';

import LogoLink from 'components/LogoLink';
import { selectReviewCount } from 'containers/App/selectors';

import OnCanvasMenu from './OnCanvasMenu';
import OffCanvasToggle from './OffCanvasToggle';
import OffCanvasMenu from './OffCanvasMenu';
import { Header, Nav } from './styles';

class SiteHeader extends React.Component {
  static propTypes = {
    reviewCount: PropTypes.number,
  };

  static defaultProps = {
    reviewCount: 0,
  };

  state = {
    headerHeight: 70, // ballpark fallback
    offCanvasMenuActive: false,
    offCanvasToggleVisible: false,
  };

  componentDidMount() {
    this.handleResize();
    window.addEventListener('resize', this.handleResize);
    window.addEventListener('click', this.hideOffCanvasMenu);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
    window.removeEventListener('click', this.hideOffCanvasMenu);
  }

  handleResize = debounce(() => {
    this.setState(() => ({
      headerHeight: this.HeaderRef.clientHeight,
      offCanvasToggleVisible: window.innerWidth <= breakpoints.md,
    }));
  }, 150);

  hideOffCanvasMenu = () => {
    if (this.state.offCanvasMenuActive === true) {
      this.setState({ offCanvasMenuActive: false });
    }
  }

  handleToggle = (event) => {
    event.stopPropagation();
    this.setState((prevState) => ({
      offCanvasMenuActive: !prevState.offCanvasMenuActive,
    }));
  }

  handleLogout = () => {
    clearToken();
    this.props.history.push('/logout'); // eslint-disable-line react/prop-types
  }

  render() {
    const isWideViewport = !this.state.offCanvasToggleVisible;
    let onCanvasRoutes = [
      // { text: 'lessons', count: this.props.lessonCount },
      { text: 'reviews', count: this.props.reviewCount },
      { text: 'vocabulary' },
    ];

    let offCanvasRoutes = [
      { text: 'settings' },
      { text: 'about' },
      { text: 'contact' },
      { text: 'logout' },
    ];

    // show all routes in main menu if large screen
    if (isWideViewport) {
      onCanvasRoutes = onCanvasRoutes.concat(offCanvasRoutes);
      offCanvasRoutes = [];
    }

    return (
      <Header innerRef={(node) => { this.HeaderRef = node; }}>
        <Nav>
          <LogoLink />
          <OnCanvasMenu
            routes={onCanvasRoutes}
            handleLogout={this.handleLogout}
          />
          <OffCanvasToggle
            isVisible={this.state.offCanvasToggleVisible}
            isActive={this.state.offCanvasMenuActive}
            handleToggle={this.handleToggle}
            ariaControls="offCanvasMenu"
          />
          <OffCanvasMenu
            id="offCanvasMenu"
            routes={offCanvasRoutes}
            offsetTop={this.state.headerHeight}
            isVisible={this.state.offCanvasMenuActive}
            handleLogout={this.handleLogout}
          />
        </Nav>
      </Header>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  reviewCount: selectReviewCount,
});

export default connect(mapStateToProps)(SiteHeader);
