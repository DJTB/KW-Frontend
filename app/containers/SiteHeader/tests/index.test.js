import 'jest-styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'enzyme';
import { createMockStore } from 'redux-logic-test';
import { MemoryRouter } from 'react-router-dom';
import { initialState } from 'shared/reducers';
import SiteHeader from '../index';

describe('<SiteHeader />', () => {
  let store;

  beforeEach(() => {
    store = createMockStore({ initialState: { ...initialState, route: { location: '' } } });
  });

  it('should match baseline snapshot', () => {
    const component = render(<Provider store={store}><MemoryRouter><SiteHeader /></MemoryRouter></Provider>);
    expect(component).toMatchSnapshot();
  });
});
