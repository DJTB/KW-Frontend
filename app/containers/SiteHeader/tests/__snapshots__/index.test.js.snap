// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteHeader /> should match baseline snapshot 1`] = `
.c2 {
  display: block;
  position: relative;
  vertical-align: middle;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 4.5rem;
  height: 4.5rem;
  background-repeat: no-repeat;
  color: #30c0f9;
  cursor: pointer;
  -webkit-transition: color .4s cubic-bezier(0.445,0.050,0.550,0.950);
  transition: color .4s cubic-bezier(0.445,0.050,0.550,0.950);
}

.c2:hover {
  -webkit-transition: color .5s cubic-bezier(0.250,0.460,0.450,0.940);
  transition: color .5s cubic-bezier(0.250,0.460,0.450,0.940);
  color: #8523e7;
}

.c2:active {
  -webkit-transition: color .1s cubic-bezier(0.250,0.460,0.450,0.940);
  transition: color .1s cubic-bezier(0.250,0.460,0.450,0.940);
  color: #ff00c3;
}

.c3 {
  display: block;
  position: absolute;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c3 .bg {
  color: inherit;
  fill: currentColor;
}

.c8 {
  text-decoration: none;
  color: inherit;
  cursor: pointer;
}

.c10 {
  position: relative;
  -webkit-transform: translateY(-4px);
  -ms-transform: translateY(-4px);
  transform: translateY(-4px);
  padding-bottom: 4px;
  bottom: -4px;
}

.c10:not(:only-child) {
  padding-right: .4em;
}

.c10:after {
  display: block;
  position: absolute;
  content: "";
  width: 0;
  bottom: 0;
  left: 50%;
  opacity: 0;
  height: 3px;
  background: #890ce9;
  -webkit-transition: all .3s ease-out;
  transition: all .3s ease-out;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c7 {
  padding: 0.39375rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.25em;
  font-weight: 600;
  font-variant: small-caps;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  color: currentColor;
}

.c7:hover .c9:after,
.c7:focus .c9:after,
.c7:active .c9:after,
.c7.active .c9:after {
  opacity: 1;
  width: 100%;
}

.c21 {
  background: none;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  border-radius: 0;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
  -webkit-user-select: none !important;
  -moz-user-select: none !important;
  -ms-user-select: none !important;
  padding: 0.39375rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.25em;
  font-weight: 600;
  font-variant: small-caps;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  color: currentColor;
}

.c21::-moz-focus-inner {
  border: 0 !important;
  padding: 0 !important;
}

.c21:hover .c9:after,
.c21:focus .c9:after,
.c21:active .c9:after,
.c21.active .c9:after {
  opacity: 1;
  width: 100%;
}

.c5 {
  font-size: 15.3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  max-height: 70px;
}

.c19 {
  font-size: 15.3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  max-height: 70px;
  font-size: 23px;
}

.c19 .c6,
.c19 .c20 {
  width: 100%;
  height: 100%;
}

.c19 .c6:hover:not(.active) > .c9:after,
.c19 .c20:hover:not(.active) > .c9:after {
  opacity: 0;
  width: 0;
}

.c19 .c9 {
  bottom: 0;
}

.c0 {
  padding-left: calc(50vw - 650px);
  padding-right: calc(50vw - 650px);
  margin-bottom: 0.39375rem;
  grid-area: Header;
  position: relative;
  background: linear-gradient(to bottom,#f0f0f0,#fafafa);
  box-shadow: 0 1px 10px rgba(59,59,59,0.1);
}

.c1 {
  padding-left: 0.39375rem;
  padding-right: 0.39375rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  list-style: none;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  width: 2.5em;
  height: 2.5em;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c17 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  width: 3rem;
  height: 3rem;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c14 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  fill: currentColor;
}

.c12 {
  padding: 0.225rem;
  font-size: 15.3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  min-width: 8em;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  color: #fafafa;
  border: 2px solid #4c81d6;
  background-color: #4c81d6;
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
  transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955);
}

.c12:focus {
  outline: none;
}

.c12:focus {
  outline: none;
}

.c12 {
  background: none;
  border: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  overflow: visible;
  padding: 0;
  border-radius: 0;
  -webkit-appearance: none !important;
  -moz-appearance: none !important;
  appearance: none !important;
  -webkit-user-select: none !important;
  -moz-user-select: none !important;
  -ms-user-select: none !important;
  min-width: auto;
}

.c12::-moz-focus-inner {
  border: 0 !important;
  padding: 0 !important;
}

.c11 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c15 {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  right: 0;
  background: rgba(133,35,231,0.95);
  color: #fafafa;
  z-index: 10;
  visibility: hidden;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  -webkit-transition: -webkit-transform 375ms cubic-bezier(0.455,0.030,0.515,0.955),visibility 0s 0.5s;
  -webkit-transition: transform 375ms cubic-bezier(0.455,0.030,0.515,0.955),visibility 0s 0.5s;
  transition: transform 375ms cubic-bezier(0.455,0.030,0.515,0.955),visibility 0s 0.5s;
}

.c18 {
  list-style: none;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  margin-top: 4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  position: absolute;
  right: .8rem;
  top: .8rem;
  overflow: hidden;
  z-index: 100;
}

@media (min-width:600px) {
  .c7 {
    padding: 0.6665rem;
  }
}

@media (min-width:600px) {
  .c21 {
    padding: 0.6665rem;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c5 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c5 {
    font-size: 17.4px;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c19 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c19 {
    font-size: 17.4px;
  }
}

@media (min-width:400px) and (max-width:2000px) {
  .c19 {
    font-size: calc(23px + (52 - 23) * ( (100vw - 400px) / ( 2000 - 400) ));
  }
}

@media (min-width:2000px) {
  .c19 {
    font-size: 52px;
  }
}

@media (min-width:600px) {
  .c0 {
    margin-bottom: 0.6665rem;
  }
}

@media (min-width:600px) {
  .c1 {
    padding-left: 0.6665rem;
  }
}

@media (min-width:600px) {
  .c1 {
    padding-right: 0.6665rem;
  }
}

@media (min-width:600px) {
  .c12 {
    padding: 0.33325rem;
  }
}

@media (min-width:600px) and (max-width:1800px) {
  .c12 {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width:1800px) {
  .c12 {
    font-size: 17.4px;
  }
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <a
      class="c2"
      href="/"
      title="Return Home"
    >
      <image_mock
        class="c3"
        height="100%"
        width="100%"
      />
    </a>
    <div
      style="margin:0 auto 0 .5rem"
    >
      <ul
        class="c4"
      >
        <li
          class="c5"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/lessons"
          >
            <span
              class="c9 c10"
            >
              lessons
            </span>
          </a>
        </li>
        <li
          class="c5"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/reviews"
          >
            <span
              class="c9 c10"
            >
              reviews
            </span>
          </a>
        </li>
      </ul>
    </div>
    <button
      aria-label="Open menu"
      class="c11 c12"
      color="#fafafa"
      type="button"
    >
      <span
        class="c13"
        color="currentColor"
      >
        <svg
          class="c14"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          title="MENU"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      class="c15"
    >
      <button
        aria-label="Close menu"
        class="c16 c12"
        color="#fafafa"
        type="button"
      >
        <span
          class="c17"
          color="currentColor"
        >
          <svg
            class="c14"
            height="100%"
            preserveAspectRatio="xMidYMid meet"
            title="CLOSE"
            viewBox="0 0 24 24"
            width="100%"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
      </button>
      <ul
        class="c18"
      >
        <li
          class="c19"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/vocabulary/levels"
          >
            <span
              class="c9 c10"
            >
              vocabulary
            </span>
          </a>
        </li>
        <li
          class="c19"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/settings"
          >
            <span
              class="c9 c10"
            >
              settings
            </span>
          </a>
        </li>
        <li
          class="c19"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/about"
          >
            <span
              class="c9 c10"
            >
              about
            </span>
          </a>
        </li>
        <li
          class="c19"
          title=""
        >
          <a
            aria-current="false"
            class="c6 c7 c8"
            href="/contact"
          >
            <span
              class="c9 c10"
            >
              contact
            </span>
          </a>
        </li>
        <li
          class="c19"
          title=""
        >
          <button
            class="c20 c21"
          >
            <span
              class="c9 c10"
            >
              logout
            </span>
          </button>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
